### PR TITLE
Add tests for LabelModel

### PR DESCRIPTION
**Pull request by paroma**
* Add tests for scheduler and optimizer

* Add tests for model load and save

Stacked PR with #1244

### CODE REVIEW

File: test/labeling/model/test_label_model.py [hunk 1]
```diff
@@ -245,6 +245,38 @@ def test_train_model(self):
...
+            ValueError, "Did not recognize lr_scheduler option"
+        ):
+            label_model.train_model(L, n_epochs=1, lr_scheduler="bad_scheduler")
+
+    def test_save_and_load(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1, lr_scheduler=None)
+        label_model.save("./test_save")
```
Comment (bhancock8) [inline test/labeling/model/test_label_model.py]: Use tempfile so there isn't a file left after the test runs. Search the other tests for some examples (sometimes creating a new directory, sometimes a new file).

Comment (paroma) [reply test/labeling/model/test_label_model.py]: done!

File: test/labeling/model/test_label_model.py [hunk 2]
```diff
@@ -245,6 +247,42 @@ def test_train_model(self):
...
+        ):
+            label_model.train_model(L, n_epochs=1, lr_scheduler="bad_scheduler")
+
+    def test_save_and_load(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1, lr_scheduler=None)
+        fd, path = tempfile.mkstemp()
+        try:
```
Comment (henryre) [inline test/labeling/model/test_label_model.py]: Shouldn't really have try blocks in tests

File: test/labeling/model/test_label_model.py [hunk 3]
```diff
@@ -245,6 +247,42 @@ def test_train_model(self):
         next_loss = label_model.loss_mu().detach().numpy().ravel()[0]
 
         self.assertLessEqual(next_loss, init_loss)
+        with self.assertRaisesRegex(
+            Exception, "Loss is NaN. Consider reducing learning rate"
```
Comment (henryre) [inline test/labeling/model/test_label_model.py]: Nit: if you want to save on lines, you can do snippets from the exception

```python
with self.assertRaisesRegex(Exception, "Loss is NaN"):
    ...
```

File: test/labeling/model/test_label_model.py [hunk 4]
```diff
@@ -245,6 +247,42 @@ def test_train_model(self):
...
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1, lr_scheduler=None)
+        fd, path = tempfile.mkstemp()
+        try:
+            label_model.save(path)
+            label_model.load(path)
+        finally:
+            os.remove(path)
```
Comment (henryre) [inline test/labeling/model/test_label_model.py]: does this work correctly? usually do shutil.rmtree

Comment (paroma) [reply test/labeling/model/test_label_model.py]: works, but switched to `shutil` to remove `try: finally:`
